test/functional/wallet_pruning.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python3
+# Copyright (c) 2022 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+"""Test wallet import on pruned node."""
+import os
+
+from test_framework.util import assert_raises_rpc_error
+from test_framework.blocktools import create_coinbase
+from test_framework.messages import CBlock
+from test_framework.script import (
+    CScript,
+    OP_NOP,
+    OP_RETURN,
+)
+from test_framework.wallet_util import get_key
+from test_framework.test_framework import BitcoinTestFramework
+
+
+class WalletPruningTest(BitcoinTestFramework):
+    nTime = 0
+    def mine_large_blocks(self, node, n):
+        # Taken from test/functional/feature_pruning.py
```

Comment:
If we're reusing helper functions, perhaps we should extract them to a single place in a follow-up PR.